contracts/Splitter.sol

```diff
@@ -2,39 +2,49 @@
 pragma solidity >= 0.5.0 < 0.7.0;
 
 contract Splitter {
-    
-    address sender = 0xA2aDe56e2c69589eaFA636C845b296718D5766fd; //Alice
-    
-    address payable[] internal recipients = [0x20601F1Ddb44E28b1511EdBD316A368D80116408, //"Bob";
-                                            0x24F1500890505ceD8534502ab403F20ce99feea0]; //"Carol";
-
-    event Deposit(uint date,
-                  address indexed _from,
-                  address indexed _to,
-                  uint amount);
-
-    function deposit() external payable{
-        doDeposit(msg.sender, msg.value);
-    }
 
-    function doDeposit(address _sender, uint _value) public payable {
-        if(_sender != sender){
-            revert("Only Alice can send ether");
-        }
-        
-        uint split = split(_value);
-        
-        for (uint i=0; i < recipients.length; i++) {
-            recipients[i].transfer(split);
+    mapping(address => uint) public balances;
+
+    event Deposit(
+        uint date,
+        address indexed sender,
+        address indexed recipient1,
+        address indexed recipient2,
+        uint amount,
+        uint remainder
+    );
+
+    event WithDraw(
+        address indexed withdrawer,
+        uint amount
+    );
+
+    function splitDeposit(address payable recipient1, address payable recipient2) external payable{
+        require(msg.value > 0, "The value must be greater than 0");
+        require(recipient1 != recipient2, "The first recipient is the same as the second recipient");
+        require(msg.sender != recipient1 && msg.sender != recipient2, "The sender cannot also be a recipient");
+
+        uint split = msg.value / 2;
 
-            emit Deposit(now, _sender, recipients[i], split);
+        balances[recipient1] += split;
+        balances[recipient2] += split;
```

Comment:
You ought to use `SafeMath` to protect against `uint` overflow. Admittedly unlikely in the case of Ether balances, but still.